resmem.c

```diff
@@ -50,37 +52,295 @@ static inline int get_info_infile(char *fname, char *res, void *out)
 	return 0;
 }
 
+/* read information from a cgroup file.
+ */
+static inline size_t cgmemread(char *cg, char *file)
+{
+	char buf[MEMBUF_128];
+	char fn[FNAMELEN];
+
+	snprintf(fn, FNAMELEN, "%s/%s%s/%s", DEFAULTCGFS,
+		MEMCGNAME, cg, file);
+
+	if (file_to_buf(fn, buf, MEMBUF_128) == -1)
+		return 0;
+
+	return (strtoul(buf, NULL, 10) / 1024);
```

Comment:
Hm, you probably want something more sane than this that does proper overflow and other error handling? Not sure. If you do we have a couple of helpers you can probably copy in LXC. See:
https://github.com/lxc/lxc/blob/41cd8a8d8c3e5fd357a4b2fcdceb7b97d70586a4/src/lxc/string_utils.c#L642

Reply:
yeah that makes sense, will use it.